src/main/java/de/tum/in/www1/artemis/web/rest/AthenaResource.java

```diff
@@ -79,17 +79,17 @@ public AthenaResource(CourseRepository courseRepository, TextExerciseRepository
     }
 
     @FunctionalInterface
-    private interface FeedbackProvider<ExerciseType, SubmissionType, OutputType> {
+    private interface FeedbackProvider<ExerciseType, SubmissionType, Boolean, OutputType> {
```

Comment:
Adding `Boolean` to the generic type declarations does not really make sense. It has always the type `Boolean` anyways and overloads said default java name.